src/components/PostDetails.tsx

```diff
@@ -1,106 +1,120 @@
-import React from 'react';
+import React, { useEffect, useState } from 'react';
 import { Loader } from './Loader';
+import { Post } from '../types/Post';
+import { Comment } from '../types/Comment';
+import { deleteComment } from '../api/comments';
 import { NewCommentForm } from './NewCommentForm';
 
-export const PostDetails: React.FC = () => {
+type Props = {
+  selectedPost: Post | null;
+  comments: Comment[];
+  setComments: (comment: Comment[]) => void;
+  loading: boolean;
+  isError: string;
+};
+
+export const PostDetails: React.FC<Props> = ({
+  selectedPost,
+  comments,
+  setComments,
+  loading,
+  isError,
+}) => {
+  const [form, setForm] = useState(false);
+
+  useEffect(() => {
+    setForm(false);
+  }, [selectedPost]);
+
+  function deleteCurrComment(commentId: number) {
+    const updatedComments = comments.filter(com => com.id !== commentId);
+
+    setComments(updatedComments);
+
+    return deleteComment(commentId).catch(error => {
+      setComments(comments);
+      throw error;
+    });
+  }
+
+  function addNewComment(newComment: Comment) {
+    setComments([...comments, newComment]);
+  }
+
   return (
     <div className="content" data-cy="PostDetails">
       <div className="content" data-cy="PostDetails">
```

Comment:
The `className` and `data-cy` attributes for the main content div are duplicated. Consider removing one of them to avoid redundancy.